examples/zookeeper/template.json

```diff
@@ -0,0 +1,265 @@
+    {
+      "kind": "Service",
+      "apiVersion": "v1beta3",
+      "metadata": {
+        "name": "zookeeper-2",
+        "labels": {
+          "name": "zookeeper-2"
+        }
+      },
+      "spec": {
+        "ports": [
+          {
+            "name": "client",
+            "port": 2181,
+            "targetPort": 2181
+          },
+          {
+            "name": "followers",
+            "port": 2888,
+            "targetPort": 2888
+          },
+          {
+            "name": "election",
+            "port": 3888,
+            "targetPort": 3888
+          }
+        ],
+        "selector": {
+          "server-id": "2"
+        }
+      }
+    },
+    {
+      "kind": "Service",
+      "apiVersion": "v1beta3",
+      "metadata": {
+        "name": "zookeeper-3",
+        "labels": {
+          "name": "zookeeper-3"
+        }
+      },
+      "spec": {
+        "ports": [
+          {
+            "name": "client",
+            "port": 2181,
+            "targetPort": 2181
+          },
+          {
+            "name": "followers",
+            "port": 2888,
+            "targetPort": 2888
+          },
+          {
+            "name": "election",
+            "port": 3888,
+            "targetPort": 3888
+          }
+        ],
+        "selector": {
+          "server-id": "3"
+        }
+      }
+    },
+    {
+      "kind": "Pod",
+      "id": "zookeeper-1",
+      "apiVersion": "v1beta1",
+      "labels": {
+        "name": "zookeeper",
+        "server-id": "1"
+      },
+      "desiredState": {
+        "manifest": {
+          "id": "zookeeper-1",
+          "containers": [
+            {
+              "name": "server",
+              "image": "${ZOOKEEPER_IMAGE}",
+              "env":[
+                { "name": "SERVER_ID", "value": "1" }
+              ],
+              "ports":[
+                {
+                  "containerPort": 2181
+                },
+                {
+                  "containerPort": 2888
+                },
+                {
+                  "containerPort": 3888
+                }
+              ]
+            }
+          ]
+        }
+      }
+    },
+    {
+      "kind": "Pod",
+      "id": "zookeeper-2",
+      "apiVersion": "v1beta1",
+      "labels": {
+        "name": "zookeeper",
+        "server-id": "2"
+      },
+      "desiredState": {
+        "manifest": {
+          "id": "zookeeper-2",
+          "containers": [
+            {
+              "name": "server",
+              "image": "${ZOOKEEPER_IMAGE}",
+              "env":[
+                { "name": "SERVER_ID", "value": "2" }
+              ],
+              "ports":[
+                {
+                  "containerPort": 2181
+                },
+                {
+                  "containerPort": 2888
+                },
+                {
+                  "containerPort": 3888
+                }
+              ]
+            }
+          ]
+        }
+      }
+    },
+    {
+      "kind": "Pod",
```

Comment:
> On May 6, 2015, at 12:49 AM, Brian Grant <notifications@github.com> wrote:
> 
> In examples/zookeeper/template.json:
> 
> > +                  "containerPort": 2181
> > +                },
> > +                {
> > +                  "containerPort": 2888
> > +                },
> > +                {
> > +                  "containerPort": 3888
> > +                }
> > +              ]
> > +            }
> > +          ]
> > +        }
> > +      }
> > +    },
> > +    {
> > +      "kind": "Pod",
> I misinterpreted your first response. IIUC, it's not that the SERVER_ID value would be wrong, but that the restarted server would have no state (unless it used a persistent volume).
> 
Sorry, yes.  And in zookeeper <3.5 you must reform the exact same membership by identity to continue.
> —
> Reply to this email directly or view it on GitHub.
>